### PR TITLE
Add worldchain safes

### PR DESCRIPTION
**Pull request by danielpartida**
### Update!
Please build spells in the proper [subproject](../dbt_subprojects/) directory. For more information, please see the main [readme](../README.md), which also links to a GH discussion with the option to ask questions.

### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** [Description of the bug fix]
- **Steps to reproduce:** [How to reproduce the bug]
- **Implementation details:** [Information on how the bug was fixed]
- **Test instructions:** [How to test the fix]
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

**Comment by jeff-dude** [comment]
thanks for helping expand the chains here 🙌 
i don't believe the tables are ready yet, so i will mark as blocked / WIP for now

**Comment by danielpartida** [comment]
Sounds good @jeff-dude. I just prepared the spell to be ready once the decoded schemas + raw tables are ready

**Comment by danielpartida** [comment]
@jeff-dude fyi I implemented the base sources of Worldchain, we need those schemas for the safe_worldchain too.

On another note, I am receiving a strange error stating that the schema safe_worldchain_safes requires a schema safe_worldchain_singletons. But the safe_worldchain_singletons is implemented too. Do you know if I am missing something?

Screenshot 2024-10-09 at 19 14 39

File: sources/_base_sources/evm/worldchain_base_sources.yml
Comment (jeff-dude) [inline sources/_base_sources/evm/worldchain_base_sources.yml:1]: fyi @Hosuke on the sources for worldchain

File: dbt_subprojects/hourly_spellbook/models/_project/safe/worldchain/safe_worldchain_safes.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        when et.to = 0x41675C099F32341bf84BFc5382aF534df5C7461a then '1.4.1'
+        when et.to = 0x29fcB43b46531BcA003ddC8FCB67FFE91900C762 then '1.4.1L2'
+        else 'unknown'
+    end as creation_version,
+    try_cast(date_trunc('day', et.block_time) as date) as block_date,
+    CAST(date_trunc('month', et.block_time) as DATE) as block_month,
+    et.block_time as creation_time,
+    et.tx_hash
+from {{ source('worldchain', 'traces') }} et
+join {{ ref('safe_wolrdchain_singletons') }} s
```
Comment (jeff-dude) [inline dbt_subprojects/hourly_spellbook/models/_project/safe/worldchain/safe_worldchain_safes.sql:36]:
```suggestion
join {{ ref('safe_worldchain_singletons') }} s
```

Comment (jeff-dude) [reply dbt_subprojects/hourly_spellbook/models/_project/safe/worldchain/safe_worldchain_safes.sql:36]: looks like just a typo?

**Review by jeff-dude — CHANGES_REQUESTED** [review]

**Comment by jeff-dude** [comment]
looks like you have source schema names incorrect?
should be just `safe_worldchain` rather than `gnosis_` prefix?

**Comment by danielpartida** [comment]
The smart contract is developed by Safe. Hence, I would prefer if the decoded schema lives under gnosis_safe instead of safe_worldchain

**Comment by danielpartida** [comment]
@jeff-dude The smart contract is developed by Safe. Hence, I would prefer it if the decoded schema goes under gnosis_safe instead of safe_worldchain. This is also where the rest of the decoded contracts are for the other chains

**Comment by jeff-dude** [comment]
> @jeff-dude The smart contract is developed by Safe. Hence, I would prefer it if the decoded schema goes under gnosis_safe instead of safe_worldchain. This is also where the rest of the decoded contracts are for the other chains

gotcha, let me connect ya back with @agaperste then to get this cleaned up

**Comment by danielpartida** [comment]
Hi @jeff-dude, Nia just told me that the tables are ready. However, when making the build I am still facing issues. Do you know if the name of the decoded tables are correct?

**Comment by jeff-dude** [comment]
> Hi @jeff-dude, Nia just told me that the tables are ready. However, when making the build I am still facing issues. Do you know if the name of the decoded tables are correct?

can confirm it doesn't exist to your expectation:
```
select *
from gnosis_safe_worldchain.safeproxyfactory_v1_3_0_evt_proxycreation
limit 100
```

@thecaptncrunch @agaperste where do we stand on helping daniel on the safe naming standards for the new worldchain?

**Comment by agaperste** [comment]
gnosis_safe_worldchain.safeproxyfactory_v_1_3_0_evt_proxycreation

gnosis_safe_worldchain.safeproxyfactory_v_1_4_1_evt_proxycreation


these are the two tables actually -- I see a small discrepancy on the
contract name also (T_T)


started a TG chat for me, Jeff and Daniel --> let's decide what to do
quickly together

**Comment by danielpartida** [comment]
Green light! @jeff-dude @Hosuke

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @danielpartida 

[Sample safe_worldchain_safes](https://dune.com/queries/4155389)

File: models/safe/zkevm/safe_zkevm_trasactions.sql
Comment (jeff-dude) [inline models/safe/zkevm/safe_zkevm_trasactions.sql:1]: is this `zkevm` entry an accident?
it's in the old directory, so it won't run in prod anywhere. it didn't run in CI either (as expected)

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: models/safe/zkevm/safe_zkevm_trasactions.sql
Comment (danielpartida) [reply models/safe/zkevm/safe_zkevm_trasactions.sql:1]: That is odd, I will remove it in an extra commit. It should not be there. Thank you for spotting this @jeff-dude

**Review by jeff-dude — APPROVED** [review]
great, thanks!

we'll get this merged asap

**Pull request merged**